public/app/features/alerting/unified/utils/misc.ts

```diff
@@ -137,8 +137,8 @@ export function makeLabelBasedSilenceLink(alertManagerSourceName: string, labels
   return createUrl('/alerting/silence/new', silenceUrlParams);
 }
 
-export function makeDataSourceLink<T extends DataSourceJsonData>(dataSource: DataSourceInstanceSettings<T>) {
-  return createUrl(`/datasources/edit/${dataSource.uid}`);
+export function makeDataSourceLink(uid: string) {
```

Comment:
This makes it work for both `DataSourceInstanceSettings` and `DataSourceSettings` which are different data structures :)